new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Application.java

```diff
@@ -78,6 +80,42 @@ public static boolean appAccessibleInPod(
       getLogger().warning(String.format("Failed to find pod %s to check the app", podName));
       return false;
     }
-  } 
+  }
+
+  /**
+   * Check if an application is accessible inside a WebLogic server pod.
+   *
+   * @param namespace Kubernetes namespace where the WebLogic server pod is running
+   * @param podName name of the WebLogic server pod
+   * @param port internal port of the managed server running in the pod
+   * @param appPath path to access the application
+   * @param expectedResponse expected response from the app
+   * @return true if the command succeeds
+   */
+  public static boolean appAccessibleInPodKubectl(
+      String namespace,
+      String podName,
+      String port,
+      String appPath,
+      String expectedResponse
+  ) {
+
+    // calling "kubectl exec" command to access the app inside a pod
+    String cmd = String.format(
+        "kubectl -n %s exec -it %s -- /bin/bash -c 'curl http://%s:%s/%s'",
+        namespace,
+        podName,
+        podName,
```

Comment:
The podName can differ from the service name.  Should the second 'podName' be converted to DNS 1123?  Alternatively, it could  be changed to "$SERVICE_NAME" as I assume the bash command will have access to the pod's env vars...

Reply:
(DNS 1123 == podName all lower case with "_" converted to "-")

Reply:
I totally agree that the code should convert the podName to a legal DNS1123 name. But this pattern is used everywhere in the integration test cases; the assumption is that the test code will not use any non-legal-1123 name as the domainUID, clusterName, managedServerPrefix and etc.  I know this assumption may not be true once more engineers start to write integration test cases. 
Do you want to have a task to do a global fix?  We need to get this change in to fix the hanging issue that impacts everyone that run Jenkins jobs to verify their changes.

Reply:
Didn't know that.